kratos/includes/properties.h

```diff
@@ -278,31 +293,379 @@ class Properties : public IndexedObject
         return result_key;
     }
 
+    /**
+     * @brief This method returns the number of subproperties
+     * @return The current number of subproperties
+     */
+    std::size_t NumberOfSubproperties()
+    {
+        return mSubPropertiesList.size();
+    }
+
+    /**
+     * @brief This method insert a new property into the list of subproperties
+     * @param pNewSubProperty The new property to be added
+     */
+    void AddSubProperties(Properties::Pointer pNewSubProperty)
+    {
+        mSubPropertiesList.insert(mSubPropertiesList.begin(), pNewSubProperty);
+    }
```

Comment:
I would add a debug check to see if other properties with the same ID is there.

Reply:
Done